keanu-project/src/test/java/io/improbable/keanu/tensor/generic/SimpleTensorTest.java

```diff
@@ -12,6 +20,38 @@
         A, B, C, D
     }
 
+    @Test
+    public void whatDoesSliceDo() {
+        double[] data = new double[]{
+            1, 2, 3, 4, 5, 6, 7, 8,
+            5, 6, 7, 8, 9, 10, 11, 12,
+            8, 7, 6, 5, 4, 3, 2, 1,
+            4, 3, 2, 1, 0, -1, -2, -3
+        };
+
+        int[] shape = new int[]{2, 2, 2, 2, 2};
+        DataBuffer buffer = Nd4j.createBuffer(data);
+        INDArray rank4 = Nd4j.create(buffer, shape);
+
+        INDArray plucked = pluck(rank4, 0, 1);
+        assertArrayEquals(new int[]{2, 2, 2}, plucked.shape());
+        assertArrayEquals(new double[]{5, 6, 7, 8, 9, 10, 11, 12}, plucked.data().asDouble(), 0.0);
+    }
+
+
+    private INDArray pluck(INDArray from, int... indices) {
```

Comment:
This pluck function exists in the DualNumber class already right?

Reply:
Oops, meant to delete!